File: typescript/base-hmi-types/src/screens/shapes/HmiLine.ts
import { HmiProperty } from "../base.js";
import { HmiSurfaceShapeBase } from "./HmiSurfaceShapeBase.js";
import { HmiObjectType } from "../base/HmiObjectType.js";
import { HmiPointCoordinateSpace } from "./HmiPointCoordinateSpace.js";

export class HmiLine extends HmiSurfaceShapeBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiLine;
  }

  x1?: HmiProperty<number>;
  y1?: HmiProperty<number>;
  x2?: HmiProperty<number>;
  y2?: HmiProperty<number>;
  pointCoordinateSpace = HmiPointCoordinateSpace.ItemRelative;
}

File: src/BaseHmiTypes/Screens/Shapes/HmiLine.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Shapes;

public class HmiLine : HmiSurfaceShapeBase
{
    public HmiLine()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiLine;
    }

    public HmiProperty<double>? X1 { get; set; }

    public HmiProperty<double>? Y1 { get; set; }

    public HmiProperty<double>? X2 { get; set; }

    public HmiProperty<double>? Y2 { get; set; }
}
